lib/hashes.md

```diff
@@ -5,40 +5,48 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
 
 # What is a Hash?
 
-Replace me with your response...
+A hash is sort of like a dictionary. Arrays are just a numbered list of items, but a hash is a list of items and their associated values (like words in a dictionary PLUS their definitions) and those items are recalled according to their names or “keys,” instead of by what numbered position they are on a list.
 
 # What are some examples of information that would be Hashes as opposed to some other data type?
 
-Replace me with your response...
+Anything that doesn’t work as just a numbered list. A dictionary wouldn’t, of course, because you’re not going to remember that “oligarch” is the 23,485th entry in the dictionary, you’re just going to remember the word oligarch (the key) and retrieve the definition from there. Your example is helpful, too – making a list of things you need to buy is not useful if they’re coming from multiple stores, but putting them in a hash with each store as key works really well for organizing that kind of two-layered (I guess you could call it that?) data. Makes me wonder if there are further structures to use in Ruby that allow for more than two layers of data, or if we’ll learn how to further index layers within hashes, if that makes any sense (like if you’re going to break up your shopping list according to stores AND sections of each store). Exciting!
```

Comment:
It's _totally_ exciting! We will be covering a final data structure later called an 'object', but there isn't much else besides what you've now learned :)

It is nice to know that you can nest hashes and arrays to your heart's content:

```ruby
{
  "Dan" => {
    "full_name" => "Daniel",
    "age" => 550,
    "cats" => ["Miles", "Booboo"],
    "best_games_of_tetris" => {
      53100 => "December 22, 1993",
      52991 => "October 4, 1993"
    }
  },
  "Sumeet" => {
    # you get the idea...
  }
}
```